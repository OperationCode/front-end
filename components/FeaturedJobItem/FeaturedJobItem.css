.job {
  padding: 1rem 0;
}

.link {
  color: #1a0dab;
}

.details {
  display: flex;
  font-size: 1.125rem;
  color: var(--grayDark);
  margin-top: 0.25rem;
}

.detailsContainer {
  margin-right: 1rem;
}

.detail {
  margin-left: 4px;
}

.description {
  margin: 0;
}

.remote {
  composes: detail;
  font-style: italic;
}

.icon {
  fill: var(--grayDark);
  height: 0.825rem;
}
