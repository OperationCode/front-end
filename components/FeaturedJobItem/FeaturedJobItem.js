import React from 'react';
import { string, bool } from 'prop-types';
import OutboundLink from 'components/OutboundLink/OutboundLink';
import BuildingIcon from 'static/images/icons/FontAwesome/building_icon.svg';
import CloudUploadIcon from 'static/images/icons/FontAwesome/cloud_upload_icon.svg';
import MapMarkerIcon from 'static/images/icons/FontAwesome/map_marker_icon.svg';
import styles from './FeaturedJobItem.module.css';

FeaturedJobItem.propTypes = {
  title: string.isRequired,
  source: string.isRequired,
  sourceUrl: string.isRequired,
  city: string,
  state: string,
  country: string,
  description: string.isRequired,
  remote: bool,
};

FeaturedJobItem.defaultProps = {
  remote: false,
  city: '',
  state: '',
  country: '',
};

function FeaturedJobItem({ title, source, sourceUrl, city, state, country, description, remote }) {
  return (
    <article className={styles.job}>
      <OutboundLink href={sourceUrl} analyticsEventLabel={`Featured Job ${source}`}>
        <h6>{title}</h6>
      </OutboundLink>

      <div className={styles.details}>
        <div className={styles.detailsContainer}>
          <BuildingIcon className={styles.icon} />
          <span className={styles.detail}>{source}</span>
        </div>

        <div className={styles.detailsContainer}>
          {(city || state || country) && <MapMarkerIcon className={styles.icon} />}
          {city && <span className={styles.detail}>{city},</span>}
          {state && <span className={styles.detail}>{state},</span>}
          {country && <span className={styles.detail}>{country}</span>}
        </div>

        {remote && (
          <div className={styles.detailsContainer}>
            <CloudUploadIcon className={styles.icon} />
            <span className={styles.remote}>Remote</span>
          </div>
        )}
      </div>
      <p className={styles.description}>{description}</p>
    </article>
  );
}

export default FeaturedJobItem;
