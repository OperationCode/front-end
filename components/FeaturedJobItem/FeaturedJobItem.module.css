.job {
  padding: 1rem 0;
}

.job h6 {
  display: inline;
}

.details {
  display: flex;
  flex-wrap: wrap;
  font-size: 1.125rem;
  color: var(--secondary);
  opacity: 0.8;
  margin-top: 0.25rem;
}

.detailsContainer {
  margin-right: 1rem;
}

.detail {
  margin-left: 4px;
}

.description {
  margin: 0;
}

.remote {
  composes: detail;
  font-style: italic;
}

.icon {
  fill: var(--secondary);
  opacity: 0.8;
  height: 0.825rem;
}
