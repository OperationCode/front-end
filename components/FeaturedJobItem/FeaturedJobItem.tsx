// import { string, bool } from 'prop-types';
import OutboundLink from 'components/OutboundLink/OutboundLink';
import BuildingIcon from 'static/images/icons/FontAwesome/building_icon.svg';
import CloudUploadIcon from 'static/images/icons/FontAwesome/cloud_upload_icon.svg';
import MapMarkerIcon from 'static/images/icons/FontAwesome/map_marker_icon.svg';

// FeaturedJobItem.propTypes = {
//   title: string.isRequired,
//   source: string.isRequired,
//   sourceUrl: string.isRequired,
//   city: string,
//   state: string,
//   country: string,
//   description: string.isRequired,
//   remote: bool,
// };

export type FeaturedJobItemPropsType = {
  /**
   * Title of the feautured job item.
   */
  title: string;
  /**
   * Description of the featured job item.
   */
  description: string;
  /**
   * Source of the featured job item.
   */
  source: string;
  /**
   * Url for the source of the featured job item.
   */
  sourceUrl: string;
  /**
   * Applies an optional state for the featured job.
   */
  city?: string;
  /**
   * Applies an optional state for the featured job.
   */
  state?: string;
    /**
   * Applies an optional country for the featured job.
   */
  country?: string;
  /**
   * Applies an indicator that the featured job is remote.
   * @default false
   */
  remote?: boolean;
};

// FeaturedJobItem.defaultProps = {
//   remote: false,
//   city: '',
//   state: '',
//   country: '',
// };

function FeaturedJobItem({
  title,
  source,
  sourceUrl,
  city,
  state,
  country,
  description,
  remote = false,
}: FeaturedJobItemPropsType) {
  return (
    <article className="px-0 py-4">
      <OutboundLink href={sourceUrl} analyticsEventLabel={`Featured Job ${source}`}>
        <h6>{title}</h6>
      </OutboundLink>

      <div className="flex flex-wrap mt-1 text-lg text-themeSecondary opacity-80 ">
        <div className="flex items-center gap-1.5">
          <BuildingIcon className="fill-themeSecondary opacity-80 h-3.5" />
          <span className="ml-1">{source}</span>
        </div>

        <div className="flex items-center gap-1.5">
          {(city || state || country) && (
            <MapMarkerIcon className="fill-themeSecondary opacity-80 h-3.5" />
          )}
          {city && <span className="ml-1">{city},</span>}
          {state && <span className="ml-1">{state},</span>}
          {country && <span className="ml-1">{country}</span>}
        </div>

        {remote && (
          <div className="flex items-center gap-1.5">
            <CloudUploadIcon className="fill-themeSecondary opacity-80 h-3.5" />
            <span className="ml-1">Remote</span>
          </div>
        )}
      </div>
      <p className="m-0">{description}</p>
    </article>
  );
}

export default FeaturedJobItem;
