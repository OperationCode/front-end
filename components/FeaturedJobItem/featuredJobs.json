[
  {
    "title": "Software Engineer",
    "source": "Lockheed Martin",
    "sourceUrl": "https://www.lockheedmartinjobs.com/job/boulder/software-engineer/694/8061488",
    "city": "Boulder",
    "state": "Colorado",
    "country": "United States",
    "description": "We are seeking a Software Engineer to join our dynamic and growing team! The selected candidate will design, develop, and test software solutions across our contracted efforts, supporting multiple baselines in defect resolution as well as enhancements, special projects, new business initiatives, and modernization efforts. The successful candidate will be a passionate coder, someone who loves a challenge and always rises to the occasion to ensure success. They embrace modern development practices and insist that others do as well. Only a Secret Clearance is required to start, and there are multiple openings for each position.",
    "status": "active",
    "remote": false,
    "tags": ["Java", "C++"]
  }
]
