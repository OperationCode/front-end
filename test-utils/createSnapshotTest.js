/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
