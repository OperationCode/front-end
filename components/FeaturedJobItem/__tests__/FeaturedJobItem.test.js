/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import FeaturedJobItem from '../FeaturedJobItem';

describe('FeaturedJobItem', () => {
  it('should render with just required props passed', () => {
    createSnapshotTest(
      <FeaturedJobItem
        title="DevOps Engineer"
        source="GitLab"
        sourceUrl="https://about.gitlab.com/jobs/apply/?gh_jid=4055708002"
        description="This is a fake role, since GitLab always has remote."
      />,
    );
  });

  it('should render properly with all props assigned', () => {
    createSnapshotTest(
      <FeaturedJobItem
        title="Experienced React Engineer"
        source="Formidable Labs"
        sourceUrl="https://jobs.lever.co/formidable/82919058-b73c-4a02-8589-87e2433f0a90"
        city="Seattle"
        state="WA"
        country="United States"
        description="As an Experienced React Engineer, you will do React things."
        remote
      />,
    );
  });
});
