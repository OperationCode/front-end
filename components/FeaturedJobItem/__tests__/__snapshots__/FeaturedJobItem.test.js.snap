// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedJobItem should render with many props assigned 1`] = `
<article
  className="job"
>
  <a
    href="https://jobs.lever.co/formidable/82919058-b73c-4a02-8589-87e2433f0a90"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="ScreenReaderOnly"
    >
      Opens in new window
    </span>
    <h6>
      Experienced React Engineer
    </h6>
    <svg
      className="externalLinkIcon"
    />
  </a>
  <div
    className="details"
  >
    <div
      className="detailsContainer"
    >
      <svg
        className="icon"
      />
      <span
        className="detail"
      >
        Formidable Labs
      </span>
    </div>
    <div
      className="detailsContainer"
    >
      <svg
        className="icon"
      />
      <span
        className="detail"
      >
        Seattle
        ,
      </span>
      <span
        className="detail"
      >
        WA
        ,
      </span>
      <span
        className="detail"
      >
        United States
      </span>
    </div>
    <div
      className="detailsContainer"
    >
      <svg
        className="icon"
      />
      <span
        className="remote"
      >
        Remote
      </span>
    </div>
  </div>
  <p
    className="description"
  >
    As an Experienced React Engineer, you will do React things.
  </p>
</article>
`;

exports[`FeaturedJobItem should render with required props 1`] = `
<article
  className="job"
>
  <a
    href="https://about.gitlab.com/jobs/apply/?gh_jid=4055708002"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="ScreenReaderOnly"
    >
      Opens in new window
    </span>
    <h6>
      DevOps Engineer
    </h6>
    <svg
      className="externalLinkIcon"
    />
  </a>
  <div
    className="details"
  >
    <div
      className="detailsContainer"
    >
      <svg
        className="icon"
      />
      <span
        className="detail"
      >
        GitLab
      </span>
    </div>
    <div
      className="detailsContainer"
    >
      
      
      
      
    </div>
  </div>
  <p
    className="description"
  >
    This is a fake role, since GitLab always has remote.
  </p>
</article>
`;
